File: low-level-design/design-patterns/python/creational/factory-pattern/azure_storage_client.py
from storage_client import StorageClient

class AzureStorageClient(StorageClient):
    def upload(self, content: str, path: str):
        print(f"Uploading {content} to Azure at {path}")    
    def download(self, path: str):
        print(f"Downloading from Azure at {path}")

File: low-level-design/design-patterns/golang/creational/factory-pattern/factory/azure_storage_client.go
package factory

import "fmt"

func NewAzureStorageClient() *AzureStorageClient {
	return &AzureStorageClient{}
}	

type AzureStorageClient struct {
}

func (client *AzureStorageClient) Upload(fileName string, data []byte) error {
	// Implementation for uploading to Azure Storage
	fmt.Println("Uploading to Azure Storage:", fileName, string(data))
	return nil
}

func (client *AzureStorageClient) Download(fileName string) error {
	// Implementation for downloading from Azure Storage
	fmt.Println("Downloading from Azure Storage:", fileName)
	return nil
}
